test/mailers/subscription_mailer_test.rb

```diff
@@ -13,20 +13,22 @@ def setup
                         :subscription_type => 'report')
 
       @user.mail_notifications << MailNotification[:subscriptions_expiring_soon]
+      @user.user_mail_notifications.first.update(mail_query: Setting[:expire_soon_days])
 
       @pool_not_expiring_soon = FactoryBot.create(:katello_pool,
                                                   :not_expiring_soon,
+                                                  :with_organization,
                                                   cp_id: "1234",
                                                   subscription_id: ActiveRecord::FixtureSet.identify(:other_subscription))
       ActionMailer::Base.deliveries = []
     end
 
-    def setup_expiring_pool
+    def set_up_expiring_pool
```

Comment:
Instead of putting this into a method, we could use a `let` ie

```
let(:expiring_pool) do
 FactoryBot.create(....)
end
```

Lets are lazily evaluated, so the pool won't be created until `expiring_pool` is called as if it were a local variable. And calling it multiple times will return the created pool, not invoke the block multiple times. Seems like the right fit for these test scenarios.

Reply:
@jturel Thanks, updated